File: backend/src/main/java/com/agentworkbench/harness/todo/entity/SessionTodo.java
package com.agentworkbench.harness.todo.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("session_todo")
public class SessionTodo {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long sessionId;

    private String content;

    private String description;

    private String activeForm;

    private String status;

    private Integer sortOrder;

    private String owner;

    private LocalDateTime claimedAt;

    private String blockedBy;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: backend-ts/src/harness/todo/entity/session-todo.ts
export interface SessionTodo {
  id?: number;
  sessionId: number;
  content?: string | null;
  description?: string | null;
  activeForm?: string | null;
  status?: string | null;
  sortOrder?: number | null;
  owner?: string | null;
  claimedAt?: string | null;
  blockedBy?: string | null;
  deleted?: number;
  createdAt?: string | null;
  updatedAt?: string | null;
}
